[PR]Design database

sho012b opened the pull request:
# WHAT
1. Changed README File type from rdoc to md
2. Added details of database design to README

# WHY
1. To use Markdown
2. Necessary for developing database of ChatSpace

KoichiOzaki commented on README.md:
```diff
@@ -0,0 +1,31 @@
+## Users table
+|column    |type      |not null|unique|foreign key|index|
+|:---------|:---------|:------:|:----:|:---------:|:---:|
+|user_name |string    |Y       |Y     |N          |Y    |
```
userテーブルのカラムなので、カラム名称はnameで大丈夫です。

sho012b replied:
カラム名称を"name"に修正しました。

KoichiOzaki commented on README.md:
```diff
@@ -0,0 +1,31 @@
+|email     |string    |Y       |Y     |N          |N    |
+|password  |string    |Y       |Y     |N          |N    |
+
+___Association:___ has_many :groups, has_many :messages
```
GroupsテーブルとUsersテーブルの関係は多対多になるので、中間テーブルを作りましょう。

sho012b replied:
中間テーブル "users_groups"を作成し、アソシエーションを "has_and_belongs_to_many"を用いて記述しました。

KoichiOzaki commented on README.md:
```diff
@@ -0,0 +1,31 @@
+|:---------|:---------|:------:|:----:|:---------:|:---:|
+|group_name|string    |Y       |Y     |N          |N    |
+
+___Association:___ has_many :users, has_many :messages
```
Usersテーブルに指摘した点と同じく中間テーブルを作ってアソシエーションしましょう。

sho012b replied:
中間テーブル "users_groups"を作成し、アソシエーションを "has_and_belongs_to_many"を用いて記述しました。

KoichiOzaki commented on README.md:
```diff
@@ -0,0 +1,31 @@
+|body      |text      |N       |N     |N          |N    |
+|image     |string    |N       |N     |N          |N    |
+|group_id  |integer   |Y       |Y     |Y          |N    |
+|user_id   |integer   |Y       |Y     |Y          |N    |
```
group_idとuser_idにはindex貼っても良いと思います。

sho012b replied:
group_idとuser_idのindexをtrueとしました。

KoichiOzaki commented on README.md:
```diff
@@ -0,0 +1,31 @@
+## Groups table
+|column    |type      |not null|unique|foreign key|index|
+|:---------|:---------|:------:|:----:|:---------:|:---:|
+|group_name|string    |Y       |Y     |N          |N    |
```
Groupsテーブルのカラムなのでカラム名称はnameで大丈夫です。

sho012b replied:
カラム名称を"name"に修正しました。

sho012b commented:
# What
1-1. Modified _"user_name"_ to _"name"_ in Users table  
1-2. Modified _"group_name"_ to _"name"_ in Groups table
2-1. Added _Index_ to "user_id" of Messages table
2-2. Added _Index_ to "group_id" of Messages table
3-1. Modified association of Users table from _"has_many"_ to _"has_and_belongs_to_many"_
3-2. Modified association of Groups table from _"has_many"_ to _"has_and_belongs_to_many"_ 
3-3. Added Users_Groups table

# Why
1-1. "user_" is not necessary since it is users table
1-2. "group_" is not necessary since it is groups table
2-1. 
2-2.
3-1. to build many-to-many relation between groups table and users table
3-2. same as 3-1
3-3. joining table is necessary for many-to-many relation between groups table and users table

sho012b commented:
頂いたコメントを基に記述を修正しました。再度レビューをお願いいたします。

また、頂いたコメントとは別に下記の通り一点修正を加えました。
・"users table"から"password"カラムを削除（deviseのデフォルト設定で作成されるため）
